pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -146,6 +146,9 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	// Reset the `EnableStaticTokenKubeconfig` value to false, when shoot cluster is updated to  k8s version >= 1.27.
 	if versionutils.ConstraintK8sLess127.Check(oldShootKubernetesVersion) && versionutils.ConstraintK8sGreaterEqual127.Check(shootKubernetesVersion) {
 		maintainedShoot.Spec.Kubernetes.EnableStaticTokenKubeconfig = pointer.Bool(false)
```

Comment:
Also, WDYT about setting `EnableStaticTokenKubeconfig` to false only if `EnableStaticTokenKubeconfig` is true in the old shoot?

Reply:
Yes makes sense.